### DISCUSSION TITLE
Graceful error-handling with appender

### DISCUSSION THREAD

**Discussion by amoghnjoshi** [duckdb/duckdb#5122] [Q&A]
Hi,

I am using DuckDB Appender C APIs as per the example mentioned at https://duckdb.org/docs/api/c/appender. 
As per the API document, the duckdb_append_end_row() function returns DuckDBSuccess or DuckDBError.
At the moment, I am observing 'INTERNAL Error: Invalid unicode detected in segment statistics update!' error. [But there can be other errors too.] 
With these errors, the appender stops committing. It just creates a temp folder and writes data there.

So, the question is, in general, what is the preferred way to gracefully handle such appender errors? 
Is it possible to skip the problematic row and continue appending next rows without closing the appender?

I also went through https://github.com/duckdb/duckdb/issues/1650 which talks about this specific error mentioned above.
However, this discussion aims at finding out a generic way to handle appender errors without losing the already appended data.

In short, what is the way to recover when the appender goes into error state (while using C apis)?
